Prose/ProseLib/Sources/UnreadFeature/UnreadScreen.swift

```diff
@@ -61,68 +56,138 @@ public struct UnreadScreen: View {
+                            senderId: "baptiste@crisp.chat",
+                            senderName: "Baptiste",
+                            avatar: PreviewImages.Avatars.baptiste.rawValue,
+                            content: "They forgot to ship the package.",
+                            timestamp: Date() - 2_800
+                        ),
+                        MessageViewModel(
+                            senderId: "valerian@crisp.chat",
+                            senderName: "Valerian",
+                            avatar: PreviewImages.Avatars.valerian.rawValue,
+                            content: "Okay, I see. Thanks. I will contact them whenever they get back online. 🤯",
+                            timestamp: Date() - 3_000
+                        ),
+                    ]
+                ),
+                .init(
+                    chatId: .person(id: "julien@thefamily.com"),
+                    messages: [
+                        MessageViewModel(
+                            senderId: "baptiste@crisp.chat",
+                            senderName: "Baptiste",
+                            avatar: PreviewImages.Avatars.baptiste.rawValue,
+                            content: "Can I initiate a deployment of the Vue app?",
+                            timestamp: Date() - 9_000
+                        ),
+                        MessageViewModel(
+                            senderId: "julien@thefamily.com",
+                            senderName: "Julien",
+                            avatar: PreviewImages.Avatars.julien.rawValue,
+                            content: "Yes, it's ready. 3 new features are shipping! 😀",
+                            timestamp: Date() - 10_000
+                        ),
+                    ]
+                ),
+                .init(
+                    chatId: .group(id: "constellation"),
+                    messages: [
+                        MessageViewModel(
+                            senderId: "baptiste@crisp.chat",
+                            senderName: "Baptiste",
+                            avatar: PreviewImages.Avatars.baptiste.rawValue,
+                            content: "⚠️ I'm performing a change of the server IP definitions. Slight outage expected.",
+                            timestamp: Date() - 90_000
+                        ),
+                        MessageViewModel(
+                            senderId: "constellation-health@crisp.chat",
+                            senderName: "constellation-health",
+                            avatar: PreviewImages.Avatars.constellationHealth.rawValue,
+                            content: "🆘 socket-1.sgp.atlas.net.crisp.chat - Got HTTP status: \"503 or invalid body\"",
+                            timestamp: Date() - 100_000
+                        ),
+                    ]
+                ),
+            ]))
+            content(state: .init(messages: .init()))
+        }
+
+        @ViewBuilder
```

Comment:
AFAIK `@ViewBuilder` is redundant as it only applies to closures.

Reply:
Here, it is, as the result is a simple `View`. However, you can put it here if you have conditional views inside (`if`, `switch`), and Swift will understand it's a view and not throw an error saying "Your function has multiple result types".

I put this out of habit, but also keep it because _maybe_ SwiftUI does some optimisations, but on that I have no idea.

Reply:
Read/watched more on this. You're absolutely right that it not only applies to closures, as I've incorrectly implied, but wraps the function in a `ViewBuilder` resultBuilder which would allow conditional blocks.

I couldn't find _any_ indication that it would improve anything when a single view is returned.

As such I wouldn't add them explicitly when not needed, but I guess they will keep popping up here and there. There also doesn't seem to be a swiftformat or swiftlint rule for that.

Reply:
Oh, thanks for the research 🙏🏻

I was sure it worked for conditional views, as it's the way to avoid wrapping the condition in a `Group`.

I won't put other ones from now (unless I have to), and remove the ones I already put soon.